tmt/beakerlib.py

```diff
@@ -219,6 +223,10 @@ def fetch(self):
         self.require = tmt.utils.listify(library.get('require', []))
         self.recommend = tmt.utils.listify(library.get('recommend', []))
 
+        self.require = [e for e in self.require if e not in self.gathered_hlp]
```

Comment:
To speed up these calculations, I'd suggest `gathered_hlp` to be a set (checking for presence is faster than in a list). Moreover, I believe that we do not care about the order of dependencies

Reply:
Are we really concerned about performance in this case?
How much libraries do we want to handle? 100k or more?
Original code just uses lists so I just following this approach and I'm just fixing one issue, not doing refactorization here. But if speed is really crucial here, we can open another MR which aims to performance issue.

Reply:
Not really (especially due to scale), but I still think it would not hurt to use a set for gathering the names.